src/BoundaryConditions/update_boundary_conditions.jl

```diff
@@ -0,0 +1,16 @@
+using Oceananigans: boundary_conditions
+
+update_boundary_condition!(bc::Union{BoundaryCondition, Nothing}, args...) = nothing
+
+function update_boundary_condition!(bcs::FieldBoundaryConditions, field, model)
+    update_boundary_condition!(bcs.west, Val(:west), field, model)
+    update_boundary_condition!(bcs.east, Val(:east), field, model)
+    update_boundary_condition!(bcs.south, Val(:south), field, model)
+    update_boundary_condition!(bcs.north, Val(:north), field, model)
+    update_boundary_condition!(bcs.bottom, Val(:bottom), field, model)
+    update_boundary_condition!(bcs.top, Val(:top), field, model)
+    update_boundary_condition!(bcs.immersed, Val(:immersed), field, model)
+end
+
+update_boundary_condition!(fields::Union{NamedTuple, Tuple}, model) = 
+    Tuple(update_boundary_condition!(boundary_conditions(field), field, model) for field in fields)
```

Comment:
```suggestion
update_boundary_condition!(fields::NamedTuple, model) = update_boundary_condition!(values(fields), model)

function update_boundary_condition!(fields::Tuple, model)
    N = length(fields)
    ntuple(Val(N)) do n
        field = fields[n]
        bcs = boundary_conditions(field)
        update_boundary_condition!(bcs, field, model)
    end
end
```

This may not fix the problem but its a start.

Another question is whether we have to pass all of `model` to `update_boundary_conditions!`. It will probably improve compilation speed and/or type inferrability if we pass just what's needed (guessing). `model` is a huge object.

Reply:
I guess its likely to just needs the grid and model fields?

Reply:
what about clock?